chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/AuthenticationController.java

```diff
@@ -153,12 +165,66 @@ public UpdatePasswordResponse changePassword(@RequestBody UpdatePasswordRequest
         return response;
     }
 
+    @Operation(summary = "Change password.",
+            description = "Change the logged in user's password as long as the old password "
+                    + "passed in matches what is stored in the database.",
+            security = {
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.API_KEY),
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.BEARER)
+            })
+    @Deprecated
+    @DeprecatedApi(friendlyUrl = "/auth/change_password",
+            message = "This endpoint is deprecated and will be removed. Please use /auth/change-password.",
+            removalDate = "2023-10-31")
+    @RequestMapping(value = "/change_password", method = RequestMethod.POST,
+            produces = "application/json; charset=utf-8")
+    public UpdatePasswordResponse changePasswordDeprecated(@RequestBody UpdatePasswordRequest request)
+            throws UserRetrievalException, MultipleUserAccountsException {
+        UpdatePasswordResponse response = new UpdatePasswordResponse();
```

Comment:
If all this code is the same code as is in the new endpoint, is there a reason not to either extract it into a private method or just call the other public one so it's just in one place?

Reply:
It is definitely a preference thing. It does seem unlikely we'd want to change the functionality behind the endpoint either to distinguish between the two, or to make a change that wouldn't cause a need to deprecate & create a new endpoint, so I see no need to make changes here